src/consumers.ts

```diff
@@ -167,15 +167,25 @@ class SecurityEngine extends AnalysisConsumer implements DiagnosticProducer
             let diagSeverity;
 
             if (this.vulnerabilityCount == 0 && this.advisoryCount > 0) {
-                diagSeverity = DiagnosticSeverity.Information; 
+                diagSeverity = DiagnosticSeverity.Information;
             } else {
                 diagSeverity = DiagnosticSeverity.Error;
             }
 
+            let msg = `${this.context.name.value}: ${this.context.version.value}`;
+            msg += `\nKnown security vulnerability 🔑: ${this.vulnerabilityCount}`;
+            msg += `\nSecurity advisory 🔨: ${this.advisoryCount}`;
+            // Show exploit information only for Registered users
+            msg += this.exploitCount != null ? `\nExploits 💣: ${this.exploitCount}` : "";
+            msg += `\nHighest severity ⚡: ${this.highestSeverity}`;
+            // Do not show the recommendation version for Free users in case of only security advisories
+            // Show recommendation for Registered Users.
+            msg += (this.changeTo && (this.vulnerabilityCount > 0 || this.exploitCount != null)) ? `\nRecommendation 💉: use version ${this.changeTo}.` : "\nRecommendation 💉: No recommended version.";
```

Comment:
Decision of showing recommended version for security advisory should be taken care by the server and the client should simply show if it has a valid value.
```suggestion
            recommendedVersion = this.changeTo || "N/A";
            msg += `\nRecommended version 💉:  ${recommendedVersion}.`;
```

Reply:
Ohk, This will require some changes from api-side as well.

but We will need the same api-response for other integrations like Clair integration and all. So removing recommended version from api-response is ok? (We do not have this user management for all integrations as well.)

Reply:
If recommended version is designated to registered users(incase it if has Snyk pvt vulns), then the API response shouldn't have it.

Reply:
The recommendation part for the quick-fix in code-action it was taken care in LSP.
but for message now we are generating from LSP we have to make changes in api-server.

Earlier the message was generated by api and the recommendation part of message was taken care from api-server.

Reply:
Okay, please raise a ticket to fix it.

Reply:
PTAL: https://issues.redhat.com/browse/APPAI-1525